src/main/scala/pioneer/practice/Practice.scala

```diff
@@ -2,19 +2,25 @@ package pioneer.practice
 
 object Practice {
   /** given a number `n`, returns n^2^ */
-  def square(n: Int): Int = -1
+  def square(n: Int): Int = n*n
 
   /**
     * given a string `s`, returns a new string with
     * two copies of `s`, back-to-back
     */
-  def repeat(s: String): String = ""
+  def repeat(s: String): String = s + s
 
   /**
     * from http://codingbat.com/prob/p187868
     *
     * Returns true if we sleep in. We sleep in if it's not a weekday or
     * we're on vacation.
     */
-  // replace this comment with your implementation of sleepIn
+  def sleepIn(weekday: Boolean, vacation: Boolean): Boolean = {
+    if (!weekday||vacation ) {
+      true
+    } else {
+      false
+    }
+  }
```

Comment:
Nice!

Because the condition itself is a boolean, you could also write:
```
def sleepin(weekday: Boolean, vacation: Boolean): Boolean = 
   !weekday || vacation
```